### DISCUSSION TITLE
[feat] Nest multiple index paths (for tabs, etc)

### DISCUSSION THREAD

**Discussion by vitzalt** [TanStack/router#696] [Ideas]
### Describe the bug

A route with `path: '/'` cannot be nested inside a route with `path: '/'`.

A good usecase for this is if you had a homepage with tabs. The homepage would be an index route, and then one of the tabs would also be an index route. The obvious way to do this would be with two nested index routes, but this shows a console error, and the app shows a blank screen.

Of course the alternative is to use a pathless route, or to just add an `id`. But I think that intuitively you want to add it as an index route, and it seems weird that it would error.

### Your Example Website or App

https://stackblitz.com/edit/github-entwu2-qyrbic?file=src%2Fmain.tsx

### Steps to Reproduce the Bug or Issue

1. Add an index route
2. Add another index route, with the first index route as the parent
3. Observe error

### Expected behavior

I expect to be able to nest index routes in a simple usecase.

### Screenshots or Videos

<img width="565" alt="tanstack-router-index" src="https://github.com/TanStack/router/assets/142844690/8cf63eac-0372-4c02-aa30-a88e5bf2e8c2">

### Platform

- OS: MacOS
- Browser: Chrome
- Version: 0.0.1-beta.160

**Comment by tannerlinsley** [comment]
I can see why you would expect that to to work. The way the type-safe route matching works right now doesn't allow this , since both routes would have the same exact unique navigation identifier (the root path `/`). I guess it's reasonable that if you *did* do this, you would expect the types of the routes to be merged together (search params, loader data, etc)

**Comment by tannerlinsley** [comment]
Since it's possible to work around this right now, as you mentioned, by turning the wrapping route into a non-path route with `id: tabs` or something similar, I think I'd rather table this until 2.0, since it could significantly break the type/matching resolution.

**Reply by abecks** [reply]
How would you do this with file based routing?

**Comment by prasid444-niural** [comment]
@vitzalt how are you handling this? Can you please explain. Thank You

**Reply by vitzalt** [reply]
Just add an `id` to the index route
